src/CalendarInput/index.js

```diff
@@ -172,14 +173,14 @@ class CalendarInput extends Component {
     const showDateSelector = !(
       !isPeriodSelection(dateSelection)
       && closeOnSelect
-      && this.state.showDateSelector
+      && this.state.showDateSelector // eslint-disable-line react/destructuring-assignment
```

Comment:
maybe you can use destructuring and rename here and you will not  need to use eslint-disable-line 
```suggestion
          const {
      showDateSelector: currentShowDateSelect,
    } = this.state
    const showDateSelector = !(
      !isPeriodSelection(dateSelection)
      && closeOnSelect
      && currentShowDateSelect
    )
```